src/App.js

```diff
@@ -1,26 +1,131 @@
+import Menu from './components/Menu';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+
+class App extends Component {
+  state = {
+    foodList: foods,
+    filteredList: foods,
+    showForm: false,
+    todaysFood:[],
+    allCalories:[]
+  };
+
+  handleAdd = (event) => {
+    event.preventDefault();
+    console.log(event.target);
+
+    let food = event.target;
+    let newFood = {
+      name: food.name.value,
+      calories: food.calories.value,
+      quantity: 0,
+      image: food.image.value,
+    };
+
+    this.setState({
+      foodList: [...this.state.foodList, newFood],
+      filteredList: [...this.state.filteredList, newFood],
+      showForm: false,
+    });
+  };
+
+  handleFormShow = () => {
+    this.setState({
+      showForm: true,
+    });
+  };
+
+  handleSearch = (event) => {
+    event.preventDefault();
+    let search = event.target.value;
+    const { foodList } = this.state;
+
+    let searchedList = foodList.filter((food) => {
+      return food.name.toLowerCase().includes(search.toLowerCase());
+    });
+
+    this.setState({
+      filteredList: searchedList,
+    });
+  };
+
+  handleTodaysFood = (item) => {
+    console.log('add quantity clicked');
+    console.log(this.state.allCalories)
+
+    let newItem = {
+      name: item.name,
+      quantity: item.quantity,
+      calories: item.calories
+    }
+
+    this.setState({
+      todaysFood: [...this.state.todaysFood, newItem],
+      allCalories: [...this.state.allCalories, newItem.calories*newItem.quantity]
+    })
+
+
+  };
+
+  render() {
+    const { filteredList, showForm, todaysFood, allCalories } = this.state;
+
+    return (
+      <div className="app">
+        <div className="header">
+          <h1 className="headline">Nutritious and Delicious</h1>
+        </div>
+        <div className="filter">
+          {showForm ? (
+            <AddFood onAdd={this.handleAdd} />
+          ) : (
+            <button
+              className="button"
+              variant="outline-primary"
+              onClick={this.handleFormShow}
+            >
+              Add new food
+            </button>
+          )}
+
+          <input
+            className="search"
+            onChange={this.handleSearch}
+            type="text"
+            placeholder="Search"
+          ></input>
+
+          <hr></hr>
+        </div>
+        <div className="columns">
+          <div className="column">
+            {filteredList.map((food) => {
+              return <Foodbox food={food} toSum={this.handleTodaysFood}/>;
```

Comment:
remember to add a key to every item that is rendered in a loop in react. This could even be the index that comes from the parameter.

Reply:
toSum might have a different meaning than handleTodaysFood. Try to have similar names for your props than the function or element being passed. You can even use the same name if you want.